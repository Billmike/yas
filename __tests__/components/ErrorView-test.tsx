import React from 'react';
import { render } from '@testing-library/react-native';
import Error from '../../app/components/ErrorView';

test('The Error component renders without error', () => {
  const { toJSON } = render(<Error text="Error" />);
  expect(toJSON()).toMatchSnapshot();
});

test('The Error component should have the text "Request Failed"', () => {
  const errorText = 'Request Failed';
  const { getByTestId } = render(<Error text={errorText} />);
  expect(getByTestId('error').props.children).toEqual(errorText);
});
