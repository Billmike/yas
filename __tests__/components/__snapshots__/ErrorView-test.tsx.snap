// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Error component renders without error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 20,
    }
  }
>
  <Text
    testID="error"
  >
    Error
  </Text>
</View>
`;
